hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -53,6 +54,30 @@ def append_error(self, key):
     def get_errors(self):
         return self[DIAGNOSTICS_ERRORS_KEY]
 
+    def assert_diagnostics_present(self, required_keys: set) -> bool:
+        """
+        Confirm that all keys specified are present in self.miner.payload.
+
+        Args:
+            required_keys (set): Key names that should be present.
+
+        Raises:
+            MissingDiagnosticsException: If one or multiple keys are not
+                found in hotspot_data then an exception mentioning the
+                missing keys is raised.
+
+        Returns:
```

Comment:
Always returning a bool is not ideal because client should know which keys are missing. I don't like not returning anything if no keys missing because that's harder to test. As a compromise, I updated it to return the list of missing keys: https://github.com/NebraLtd/hm-pyhelper/pull/93/commits/ad660b29360ba5d20223579ad6cbefc779de5245

Reply:
Yes that's fine too.